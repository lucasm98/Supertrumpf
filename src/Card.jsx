import * as React from 'react';
import PropTypes from 'prop-types';

import './Card.css';
import Animal from './Animal';
import DarkMode from './DarkMode';

export default function Card({
  animal,
  uncovered,
  onSelectProperty,
  selectedProperty,
}) {
  const front = (
    <div className="card">
      <h1>{animal.name ? animal.name : 'Unbekannt'}</h1>
      {animal.image && (
        <img
          alt={animal.name}
          src={`${process.env.PUBLIC_URL}/${animal.image}`}
          height="200"
          width="200"
        />
      )}
      <table>
        <tbody>
          {Object.keys(Animal.properties).map((property) => {
            const animalProperty = Animal.properties[property];
            return (
              <tr
                key={property}
                className={selectedProperty === property ? 'active' : ''}
                onClick={() => {
                  onSelectProperty(property);
                }}
              >
                <td>{animalProperty.label}</td>
                <td>
                  {animal[property]}
                  &nbsp;
                  {animalProperty.unit}
                </td>
              </tr>
            );
          })}
        </tbody>
      </table>
    </div>
  );
  const back = <div className="card back" />;

  return (
    <DarkMode.Consumer>
      {darkMode => {
        const darkModeClassName = darkMode ? 'dark' : 'light';
        return (
          <div className={darkModeClassName}>{uncovered? front : back}</div>
        );
      }}
    </DarkMode.Consumer>
  )
}

Card.propTypes = {
  uncovered: PropTypes.bool.isRequired,
  animal: PropTypes.object.isRequired,
  onSelectProperty: PropTypes.func,
  selectedProperty: PropTypes.string,
};
