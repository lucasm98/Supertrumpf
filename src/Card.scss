@import './variables.scss';

.card {
    border: 1px solid black;
    border-radius: 10px;
    width: 200px;
    height: 388px;
    margin: 15px;
    padding: 15px;
    box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.25);

    &.light {
        color: $lightFrontColor;
        background-color: $lightBackground;

        &.back {
            background-color: $lightBackSide;
        }

        tr:nth-child(2n-1).active,
        .active {
            background-color: yellow;
        }
    }

    &.dark {
        color: $darkFrontColor;
        background-color: $darkBackground;

        &.back {
            background-color: $darkBackSide;
        }

        tr:nth-child(2n-1) {
            background-color: #666;
        }

        tr:nth-child(2n-1).active,
        .active {
            background-color: $darkBackSide;
        }
    }

    h1 {
        font-size: 20px;
        margin: 0 0 10px 0;
    }

    table {
        width: 100%;
        border-collapse: collapse;
    }

    tr:nth-child(2n-1) {
        background-color: #ddd;
    }

    td {
        padding: 5px 0;
    }

    img {
        border-radius: 3px;
        border: 1px solid black;
        margin-bottom: 10px;
    }
}