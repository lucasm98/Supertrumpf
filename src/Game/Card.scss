@import '../variables';

.card {
    border: 1px solid black;
    border-radius: 10px;
    width: 200px;
    height: 388px;
    margin: 15px;
    padding: 15px;
    box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.25);

    &.light {
        color: $lightFontColor;
        background-color: $lightBackground;

        &.back {
            background-color: $lightBackSide;
        }
    }

    &.dark {
        color: $darkFontColor;
        background-color: $darkBackground;
        &.back {
            background-color: $darkBackSide;
        }
    }

    h1 {
        font-size: 20px;
        margin: 0 0 10px 0;
    }

    img {
        border-radius: 3px;
        border: 1px solid black;
        margin-bottom: 10px;
    }
}
