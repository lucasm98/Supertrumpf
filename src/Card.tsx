import React, {useContext} from 'react';
import classNames from 'classnames';

import './Card.scss';
import Animal from './Animal';
import DarkMode from './DarkMode';
import CardTable from './CardTable';

interface Props {
  animal: Animal;
  uncovered: boolean;
  onSelectProperty?: (property: keyof Animal) => void;
  selectedProperty?: keyof Animal | '';
}

export default function Card ({
  animal,
  uncovered,
  onSelectProperty,
  selectedProperty,
}: Props) {
  const darkMode = useContext(DarkMode);
  const front = (
    <>
      <h1>{animal.name ? animal.name : 'Unbekannt'}</h1>
      {animal.image && (
        <img
          alt={animal.name}
          src={`${process.env.PUBLIC_URL}/${animal.image}`}
          height="200"
          width="200"
        />
      )}
      <CardTable
        animal={animal}
        onSelectProperty={onSelectProperty}
        selectedProperty={selectedProperty}
        darkMode={darkMode}
      />
    </>
  );

  const cardClasses = classNames('card', {
    back: !uncovered,
    light: !darkMode,
    dark: darkMode,
  });

  return <div className={cardClasses}>{uncovered? front : ''}</div>;
}