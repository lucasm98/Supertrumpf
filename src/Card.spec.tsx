import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";
import Animal from "./Animal";

describe("Card", () => {
  describe("Snapshots", () => {
    it("should match the snapshot", () => {
      const animal = new Animal("Elefant", "Bild", 3.3, 6000, 70, 1, 40);

      const tree = renderer
        .create(<Card animal={animal} uncovered={true} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
