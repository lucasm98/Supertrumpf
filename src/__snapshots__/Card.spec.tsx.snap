// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots should match the snapshot 1`] = `
<div
  className="card light"
>
  <h1>
    Elefant
  </h1>
  <img
    alt="Elefant"
    height="200"
    src="/Bild"
    width="200"
  />
  <table
    className="sc-dkzDqf ePGVba"
  >
    <tbody>
      <tr
        className="sc-gsnTZi iytFdT"
        onClick={[Function]}
      >
        <td
          className="sc-bczRLJ OKfSj"
        >
          Größe
        </td>
        <td
          className="sc-bczRLJ OKfSj"
        >
          3.3
           
          m
        </td>
      </tr>
      <tr
        className="sc-gsnTZi iytFdT"
        onClick={[Function]}
      >
        <td
          className="sc-bczRLJ OKfSj"
        >
          Gewicht
        </td>
        <td
          className="sc-bczRLJ OKfSj"
        >
          6000
           
          kg
        </td>
      </tr>
      <tr
        className="sc-gsnTZi iytFdT"
        onClick={[Function]}
      >
        <td
          className="sc-bczRLJ OKfSj"
        >
          Alter
        </td>
        <td
          className="sc-bczRLJ OKfSj"
        >
          70
           
          Jahre
        </td>
      </tr>
      <tr
        className="sc-gsnTZi iytFdT"
        onClick={[Function]}
      >
        <td
          className="sc-bczRLJ OKfSj"
        >
          Nachkommen
        </td>
        <td
          className="sc-bczRLJ OKfSj"
        >
          1
           
        </td>
      </tr>
      <tr
        className="sc-gsnTZi iytFdT"
        onClick={[Function]}
      >
        <td
          className="sc-bczRLJ OKfSj"
        >
          Geschwindigkeit
        </td>
        <td
          className="sc-bczRLJ OKfSj"
        >
          40
           
          km/h
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
